cmd/preflight/cli/root.go

```diff
@@ -64,10 +64,13 @@ that a cluster meets the requirements to run an application.`,
 
 	cobra.OnInitialize(initConfig)
 
-	cmd.AddCommand(VersionCmd())
+	cmd.AddCommand(util.VersionCmd())
 	cmd.AddCommand(OciFetchCmd())
 	preflight.AddFlags(cmd.PersistentFlags())
 
+	// Dry run flag should not persist between across subcommands. Adding here to avoid that
+	cmd.Flags().Bool("dry-run", false, "print the preflight spec without running preflight checks")
```

Comment:
isn't this added below in pkg/preflight/flags.go?  Not sure I understand the comment.

Reply:
> isn't this added below in pkg/preflight/flags.go? Not sure I understand the comment.

All flags in `pkg/preflight/flags.go` are also available in all subcommands e.g `preflight completions`.

```sh
[evans] $ preflight completion -h
Generate the autocompletion script for preflight for the specified shell.
See each sub-command's help for details on how to use the generated script.

Usage:
  preflight completion [command]

Available Commands:
  bash        Generate the autocompletion script for bash
  fish        Generate the autocompletion script for fish
  powershell  Generate the autocompletion script for powershell
  zsh         Generate the autocompletion script for zsh

Flags:
  -h, --help   help for completion

Global Flags:
      --collect-without-permissions   always run preflight checks even if some require permissions that preflight does not have (default true)
      --collector-image string        the full name of the collector image to use
      --collector-pullpolicy string   the pull policy of the collector image
      --cpuprofile string             File path to write cpu profiling data
      --debug                         enable debug logging
      --format string                 output format, one of human, json, yaml. only used when interactive is set to false (default "human")
      --interactive                   interactive preflights (default true)
      --memprofile string             File path to write memory profiling data
  -o, --output string                 specify the output file path for the preflight checks
      --selector string               selector (label query) to filter remote collection nodes on.
      --since string                  force pod logs collectors to return logs newer than a relative duration like 5s, 2m, or 3h.
      --since-time string             force pod logs collectors to return logs after a specific date (RFC3339)

Use "preflight completion [command] --help" for more information about a command.
```

Running `preflight completions --dry-run` or `preflight completions --since 3h` is redundant.

Reply:
Alternatively, we could change [this line](https://github.com/replicatedhq/troubleshoot/blob/d0d6f00c0f0a08d579fc4af498a007f049525d5a/cmd/preflight/cli/root.go#L69) to `preflight.AddFlags(cmd.Flags())`

Personally I prefer each subcommand to explicitly list all the flags it needs to expose